### PR TITLE
Add initial framework for experimental three way merge

### PR DESCRIPTION
**Pull request by lesaltzm**

### CODE REVIEW

File: src/PAModel/CanvasDocument.cs [hunk 1]
```diff
@@ -197,13 +197,59 @@ private static void Wrapper(Action worker, ErrorContainer errors)
...
         }
-
+                
         internal CanvasDocument()
         {
             _editorStateStore = new EditorStateStore();
             _templateStore = new TemplateStore();
         }
 
+        internal CanvasDocument(CanvasDocument other)
```
Comment (MikeStall) [inline src/PAModel/CanvasDocument.cs:207]: In theory, is this the same as 

   CanvasDocument.LoadFromMsApp(other.SaveToMsapp())

Comment (lesaltzm) [reply src/PAModel/CanvasDocument.cs:207]: Yeah, should be.

File: src/PAModel/IR/IRNode.cs [hunk 2]
```diff
@@ -33,6 +34,17 @@ public override void Accept<Context>(IRNodeVisitor<Context> visitor, Context con
         {
             visitor.Visit(this, context);
         }
+
+        public BlockNode Clone()
+        {
+            return new BlockNode()
+            {
+                Name = Name.Clone(),
+                Properties = Properties.Select(prop => prop.Clone()).ToList(),
```
Comment (MikeStall) [inline src/PAModel/IR/IRNode.cs:43]: As a nit, can we use a 
  IList<T>.Clone() extension method?

File: src/PAModel/MergeTool/Deltas/ControlPath.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.PowerPlatform.Formulas.Tools.MergeTool.Deltas
+{
+    [DebuggerDisplay("{string.Join('.', _segments)}")]
+    internal class ControlPath
```
Comment (MikeStall) [inline src/PAModel/MergeTool/Deltas/ControlPath.cs:13]: Can we comment this? 
Would this concept be useful beyond merge?

**Review by MikeStall — APPROVED** [review]

**Pull request merged**